keps/sig-apps/20190715-rollingUpdate-enhancement-for-daemonset.md

```diff
@@ -0,0 +1,182 @@
+---
+title: RollingUpdate enhancement for Daemonset
+authors:
+  - "@resouer"
+  - "@zhangxiaoyu-zidif"
+owning-sig: sig-apps
+participating-sigs:
+  - sig-apps
+reviewers:
+  - "@janetkuo"
+approvers:
+  - TBD
+editor: TBD
+creation-date: 2019-07-15
+last-updated: 2019-11-13
+status: provisional
+see-also:
+  - n/a
+replaces:
+  - n/a
+superseded-by:
+  - n/a
+---
+
+# RollingUpdate enhancement for Daemonset
+
+## Table of Contents
+
+<!-- toc -->
+- [Summary](#summary)
+- [Motivation](#motivation)
+  - [Goals](#goals)
+  - [Non-Goals](#non-goals)
+- [Proposal](#proposal)
+  - [User Stories](#user-stories)
+    - [Story 1](#story-1)
+  - [Implementation Details](#implementation-details)
+    - [API Changes](#api-changes)
+    - [Implementation](#implementation)
+  - [Risks and Mitigations](#risks-and-mitigations)
+  - [Upgrades/Downgrades](#upgradesdowngrades)
+  - [Tests](#tests)
+- [Graduation Criteria](#graduation-criteria)
+- [Implementation History](#implementation-history)
+- [Drawbacks [optional]](#drawbacks-optional)
+- [Alternatives](#alternatives)
+<!-- /toc -->
+
+## Summary
+
+The purpose of this enhancement is to enhance RollingUpdate for Daemonset.
+  - When Daemonset rolling update, it can be `Paused` by user.
+  - Add `SurgingRollingUpdate` for Daemonset as a new update strategy.
+  - Add two more fields, `partition` and `selector` for `RollingUpdate`
+    - `partition` is the number of DaemonSet pods remained to be old version.
+    - `selector` is to query over nodes whoes labels are matched by the Daemonset `RollingUpdate`.
```

Comment:
DaemonSet already has a nodeSelector which selects where to run the pods, why do we need to select a subset of those nodes by selector? What is the real use case ?